### DISCUSSION TITLE
How to get user permission without fetch all teams members?

### DISCUSSION THREAD

**Discussion by lukenvn** [appwrite/appwrite#3941] [Q&A] [answered]
Hi guys, 

I'm looking for a way to retrieve the user permission (roles) and the team which they belong to, so far what I can do are
- get all the members of a team.
- find the correct user by using userId 
- get the role from the user

Is there anybody who knows some simpler way? I know we have get specific team membership info API but it required the membershipId which I don't know how to get it without fetching all the members of the team.

Thank you in advance

**Comment by stnguyen90** [comment, marked as answer]
Hi @lukenvn! Assuming you're using a client SDK, for now, you'll need to:

1.  [List Teams](https://appwrite.io/docs/client/teams?sdk=web-default#teamsList)
2. For each Team, [List Team Memberships](https://appwrite.io/docs/client/teams?sdk=web-default#teamsListMemberships) passing the current user's user ID in the `search` parameter to filter the result to only show the current user's membership.
